pybamm/models/submodels/current_collector/quite_conductive_potential_pair.py

```diff
@@ -0,0 +1,97 @@
+#
+# Class for two-dimensional current collectors
+#
+import pybamm
+from .potential_pair import (
+    BasePotentialPair,
+    PotentialPair1plus1D,
+    PotentialPair2plus1D,
+)
+
+
+class BaseQuiteConductivePotentialPair(BasePotentialPair):
+    """A submodel for Ohm's law plus conservation of current in the current collectors,
+    which uses the voltage-current relationship from the SPM(e).
+
+    Parameters
+    ----------
+    param : parameter class
+        The parameters to use for this submodel
+
+
+    **Extends:** :class:`pybamm.current_collector.BaseModel`
+    """
+
+    def __init__(self, param):
+        super().__init__(param)
+
+    def get_fundamental_variables(self):
+
+        phi_s_cn = pybamm.standard_variables.phi_s_cn
+
+        variables = self._get_standard_negative_potential_variables(phi_s_cn)
+
+        # TODO: grad not implemented for 2D yet
+        i_cc = pybamm.Scalar(0)
+        i_boundary_cc = pybamm.standard_variables.i_boundary_cc
+
+        variables.update(self._get_standard_current_variables(i_cc, i_boundary_cc))
+
+        # Lagrangian multiplier for
+        c = pybamm.Variable("Lagrangian multiplier")
+        variables.update({"Lagrangian multiplier": c})
+
+        return variables
+
+    def set_algebraic(self, variables):
+
+        param = self.param
+        applied_current = param.current_with_time
+        cc_area = self._get_effective_current_collector_area()
+        z = pybamm.standard_spatial_vars.z
+
+        phi_s_cn = variables["Negative current collector potential"]
+        phi_s_cp = variables["Positive current collector potential"]
+        i_boundary_cc = variables["Current collector current density"]
+        i_boundary_cc_0 = variables["Leading-order current collector current density"]
+        c = variables["Lagrangian multiplier"]
+
+        self.algebraic = {
+            phi_s_cn: (param.sigma_cn * param.delta ** 2 * param.l_cn)
+            * pybamm.laplacian(phi_s_cn)
+            - pybamm.source(i_boundary_cc_0, phi_s_cn),
+            i_boundary_cc: (param.sigma_cp * param.delta ** 2 * param.l_cp)
+            * pybamm.laplacian(phi_s_cp)
+            + pybamm.source(i_boundary_cc_0, phi_s_cp)
```

Comment:
might be worth adding a comment that the second argument in source is the variable which appears in the laplacian for that eqn (i.e. the var in that eqn to which BCs are applied) but maybe this is clear from the docs for source